src/cmap/connection.ts

```diff
@@ -747,9 +747,12 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
       }
     } catch (readError) {
       if (TimeoutError.is(readError)) {
-        throw new MongoOperationTimeoutError(
+        const error = new MongoOperationTimeoutError(
           `Timed out during socket read (${readError.duration}ms)`
         );
+        this.dataEvents = null;
```

Comment:
Does this pretty much drop all packets that have been received but not read?

Reply:
Paraphrasing response from Slack
> No, I am setting it to null because the iterator has already thrown the error, connection.cleanup will call `dataEvents.throw(error)` if it exists, which it has already done because it was the one that threw
. This is just preventing us from doing the work twice.
 Typically errors come from the socket, so dataEvents.throw(e) is how you get the for-await loop to "unblock"
. It's like you're inserting the error into the iterator, this new code path is coming out of the iterator, and entering clean up, so needs a new "null it out" line of code.